browser/ui/brave_ads/ad_notification_header_view.cc

```diff
@@ -0,0 +1,148 @@
+
+#include <memory>
+
+#include "base/strings/utf_string_conversions.h"
+#include "brave/browser/ui/brave_ads/font_util.h"
+#include "brave/browser/ui/brave_ads/insets_util.h"
+#include "brave/browser/ui/brave_ads/spacer_view.h"
+#include "brave/grit/brave_generated_resources.h"
+#include "ui/accessibility/ax_enums.mojom.h"
+#include "ui/accessibility/ax_node_data.h"
+#include "ui/base/l10n/l10n_util.h"
+#include "ui/gfx/color_palette.h"
+#include "ui/gfx/font_list.h"
+#include "ui/gfx/geometry/insets.h"
+#include "ui/gfx/geometry/size.h"
+#include "ui/views/border.h"
+#include "ui/views/controls/label.h"
+#include "ui/views/layout/flex_layout.h"
+#include "ui/views/metadata/metadata_impl_macros.h"
+
+namespace brave_ads {
+
+namespace {
+
+// Spacing around each child view
+constexpr gfx::Insets kMargin(/* top */ 0,
+                              /* left */ 0,
+                              /* bottom */ 0,
+                              /* right */ 0);
+
+// Spacing between child views and host views
+constexpr gfx::Insets kInteriorMargin(/* top */ 0,
+                                      /* left */ 10,
+                                      /* bottom */ 0,
+                                      /* right */ 2);
+
+const int kHeaderViewHeight = 22;
+
+const int kControlButtonsSpacing = 10;
+
+const int kTitleFontSize = 14;
+const gfx::Font::FontStyle kTitleFontStyle = gfx::Font::NORMAL;
+const gfx::Font::Weight kTitleFontWeight = gfx::Font::Weight::SEMIBOLD;
+constexpr SkColor kTitleColor = SkColorSetRGB(0x75, 0x75, 0x75);
+const gfx::HorizontalAlignment kTitleHorizontalAlignment = gfx::ALIGN_LEFT;
+const gfx::VerticalAlignment kTitleVerticalAlignment = gfx::ALIGN_BOTTOM;
+constexpr gfx::Insets kTitleBorderInsets(/* top */ 10,
+                                         /* left */ 0,
+                                         /* bottom */ 10,
+                                         /* right */ 0);
+
+}  // namespace
+
+AdNotificationHeaderView::AdNotificationHeaderView(const int width) {
+  CreateView(width);
+}
+
+AdNotificationHeaderView::~AdNotificationHeaderView() = default;
+
+void AdNotificationHeaderView::SetTitle(const base::string16& text) {
+  DCHECK(title_label_);
+
+  title_label_->SetText(text);
+
+  NotifyAccessibilityEvent(ax::mojom::Event::kTextChanged, true);
+}
+
+void AdNotificationHeaderView::SetTitleElideBehavior(
+    gfx::ElideBehavior elide_behavior) {
+  DCHECK(title_label_);
+
+  title_label_->SetElideBehavior(elide_behavior);
+}
+
+void AdNotificationHeaderView::GetAccessibleNodeData(
+    ui::AXNodeData* node_data) {
+  node_data->role = ax::mojom::Role::kGenericContainer;
+
+  DCHECK(title_label_);
+  node_data->SetName(title_label_->GetText());
+}
+
+void AdNotificationHeaderView::OnThemeChanged() {
+  View::OnThemeChanged();
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+void AdNotificationHeaderView::CreateView(const int width) {
```

Comment:
one more nit: we never need `const` for fundamental types of function parameters